### DISCUSSION TITLE
gh gj gk 等不起作用

### DISCUSSION THREAD

**Discussion by Linya-IronMan** [nshen/learn-neovim-lua#108] [Q&A] [answered]
Copy 作者的配置代码。
gh 应该在调用 vim.lsp.buf.hover() 这个api，通过快捷键调用的时候，一点反应都没有。
所以我直接手动调用这个api
<img width="430" alt="image" src="https://user-images.githubusercontent.com/77786952/188338698-30e2da5a-213e-4854-9b3d-48572fee6cd8.png">
结果报错....，这应该是 lsp 配置有问题，毕竟这是在调用lsp的api
<img width="722" alt="image" src="https://user-images.githubusercontent.com/77786952/188338723-e50f34cc-ff83-4091-a93a-0d019cc3a8c4.png">
但是我并不知道应该如何去修复这个问题， volar coc-volar 都有进行尝试，但是并没有解决问题。

---

或许是我对这个api的理解有问题? vim.lsp.buf 指向的是哪里？

**Comment by nshen** [comment, marked as answer]
多半是lsp没启动， `:LspInfo` 看看有volar吗

lsp快捷键是 on_attach 到 buffer 时定义的，可能没调用到on_attach，所以没有快捷键。

vim.lsp.buf 应该就是当前内存中的这个文件，你自己调用的时候提示的是当前的这个buffer没有找到 lsp，所以应该是lsp没启动。

**Reply by Linya-IronMan** [reply]
![image](https://user-images.githubusercontent.com/77786952/188379033-fece3e8d-8c8a-4307-a78e-a88e4f0f40a6.png)
我已经通过 Lspinstall 安装了 volar ，但是通过 LspInfo 查看vue文件的时候，并没有找到 volar
![image](https://user-images.githubusercontent.com/77786952/188379868-6e54bc8d-bc5a-4a6d-ad17-84606ea34170.png)
Lsprestart 也执行过，也退出重新进入过

---

是不是除了 volar 这些插件之外还要自己写一些 lua 配置？我看html rust ... 在 /lua/config 下都有一个对应的文件。
但是这样的配置要怎么写？

---

大概知道on_attach 是在说什么了，需要为每个语言都写一个 lua 文件暴露一个 on_setup 接口，这个lua文件内部需要对本语言的lsp进行一些配置，决定是否开启某些功能.. 

---

 我悟了啊！！

**Reply by nshen** [reply]
嗯，别忘了在setup里引入

**Comment by HUAHUAI23** [comment]
这主题不错 问下主题名字是啥 我想试试

**Reply by Linya-IronMan** [reply]
好像是 nord
